### PR TITLE
Bug 1808282 - Fix flaky toggleSearchBookmarksAndHistoryTest UI test

### PR DESCRIPTION
**Pull request by AndiAJ**
Bug 1808282 - Fix and improve `toggleSearchBookmarksAndHistoryTest` UI test coverage ✅ successfully passed 100x on Firebase.

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [ ] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

**Review by sv-ohorvath — APPROVED** [review]

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 1]
```diff
@@ -110,31 +111,34 @@ class SearchRobot {
...
-                rule.onNodeWithTag("mozac.awesomebar.suggestions")
-                    .performScrollToNode(hasText(searchSuggestion))
-                    .assertExists()
-                break
+                for (searchSuggestion in searchSuggestions) {
+                    mDevice.waitForObjects(mDevice.findObject(UiSelector().textContains(searchSuggestion)))
+                    rule.onNodeWithTag("mozac.awesomebar.suggestions")
+                        .performScrollToNode(hasText(searchSuggestion))
+                        .assertExists()
+                }
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt:123]: Don't we still need the `break` to stop the iteration for `RETRY_COUNT`?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt [hunk 2]
```diff
@@ -110,31 +111,34 @@ class SearchRobot {
...
             } catch (e: AssertionError) {
-                if (i == RETRY_COUNT) {
-                    throw e
-                } else {
-                    expandSearchSuggestionsList()
+                mDevice.pressBack()
```
Comment (AaronMT) [inline app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt:125]: Don't we still want to check for `RETRY_COUNT` here?

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt (hunk 1 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt:123]: Yes, added the `break`.

File: app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt (hunk 2 above)
Comment (AndiAJ) [reply app/src/androidTest/java/org/mozilla/fenix/ui/robots/SearchRobot.kt]: Yes, added the check.

**Review by AaronMT — APPROVED** [review]

**Pull request merged**